core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java

```diff
@@ -644,6 +684,25 @@ private void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec,
       SqlWriter.Frame piFrame = writer.startFunCall("PI");
       writer.endFunCall(piFrame);
       break;
+    case "TO_HEX":
+      unparseUDF(writer, call, leftPrec, rightPrec, "UDF_CHAR2HEX");
```

Comment:
Can we create an enum for the last parameter and then use it to get the desired value? 
That way we can club all the other such cases together and call this method only once.

Reply:
done. created.